include/fs/fs.h

```diff
@@ -71,6 +71,11 @@ enum {
 	FS_TYPE_EXTERNAL_BASE,
 };
 
+/* Do not format volume when file system not found */
+#define FS_MOUNT_FLAG_NO_FORMAT		1
+/* Mount for read-only, this incudes FS_MOUNT_FLAG_NO_FORMAT */
+#define FS_MOUNT_FLAG_READ_ONLY		3
```

Comment:
This is a little opaque.  How about:
```
#define FS_MOUNT_FLAG_NO_FORMAT_MASK BIT(0)
#define FS_MOUNT_FLAG_READ_ONLY_MASK BIT(1)

#define FS_MOUNT_FLAG_NO_FORMAT FS_MOUNT_FLAG_NO_FORMAT_MASK
#define FS_MOUNT_FLAG_READ_ONLY (FS_MOUNT_FLAG_READ_ONLY_MASK \
                                 | FS_MOUNT_FLAG_NO_FORMAT_MASK)
```
separating the specific bits from the user-visible mount options.

Or just use the two base flags and document that read-only implies no-format.

Reply:
Will use.